ReportingServicesTools/Functions/CatalogItems/Rest/Out-RsRestFolderContent.ps1

```diff
@@ -0,0 +1,185 @@
+            }
+
+            $folder = ConvertFrom-Json $response.Content
+
+            try
+            {
+                Write-Verbose "Fetching catalog items under $RsFolder from server..."
+                $url = [string]::Format($folderCatalogItemsApiV1, $folder.Id)
+                if ($Credential -ne $null)
+                {
+                    $response = Invoke-WebRequest -Uri $url -Method Get -Credential $Credential -Verbose:$false
+                }
+                else
+                {
+                    $response = Invoke-WebRequest -Uri $url -Method Get -UseDefaultCredentials -Verbose:$false
+                }
+            }
+            catch
+            {
+                throw (New-Object System.Exception("Error while trying to fetch catalog items under $RsFolder! Exception: $($_.Exception.Message)", $_.Exception))
+            }
+        }
+        else
+        {
+            try
+            {
+                Write-Verbose "Fetching catalog items under $RsFolder from server..."
+                $url = [string]::Format($folderCatalogItemsApiLatest, $RsFolder)
+                if ($Credential -ne $null)
+                {
+                    $response = Invoke-WebRequest -Uri $url -Method Get -Credential $Credential -Verbose:$false
+                }
+                else
+                {
+                    $response = Invoke-WebRequest -Uri $url -Method Get -UseDefaultCredentials -Verbose:$false
+                }
+            }
+            catch
+            {
+                throw (New-Object System.Exception("Error while trying to fetch catalog items under $RsFolder! Exception: $($_.Exception.Message)", $_.Exception))
+            }
+        }
+
+        $catalogItems = (ConvertFrom-Json $response.Content).value
+        foreach ($catalogItem in $catalogItems)
+        {
+            if ($catalogItem.Type -eq "Report" -or
+                $catalogItem.Type -eq "DataSet" -or
+                $catalogItem.Type -eq "DataSource" -or
+                $catalogItem.Type -eq "MobileReport" -or
+                $catalogItem.Type -eq "PowerBIReport")
+            {
+                Write-Verbose "Parsing metadata for $($catalogItem.Name)..."
+                Out-RsRestCatalogItemId -RsItemInfo $catalogItem -Destination $Destination -ReportPortalUri $ReportPortalUri -RestApiVersion $RestApiVersion -Credential $Credential -WebSession $WebSession
+            }
+            elseif ($catalogItem.Type -eq "Folder" -and $Recurse)
+            {
+                $subFolderPath = "$Destination\$($catalogItem.Name)"
+                Write-Verbose "Creating folder $($catalogItem.Name)..."
+                New-Item -Path $subFolderPath -ItemType Directory
+
+                Out-RsRestFolderContent -RsFolder $catalogItem.Path -Destination $subFolderPath -ReportPortalUri $ReportPortalUri -RestApiVersion $RestApiVersion -Credential $Credential -WebSession $WebSession
+            }
```

Comment:
I would add a write-verbose with the items that aren't being exported

Reply:
Added support so don't need to do this anymore.